kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/AesGcmEncryptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+class AesGcmEncryptor {
+
+    private final SecretKey key;
```

Comment:
SecretKey implement Destroyable.  We should think about lifecycle of the SecretKey and be prepared to call destroy in keys we no longer need.

Reply:
I thought about this too. I doscovered/remembered the other day that `Destroyable` can throw `DestroyFailedException`, and for the `SecretKey` impl I was using at the time this was indeed thrown. 

I think it will be a responsibility of the DekCache (once it _is_ a cache) to somehow at least attempt to destroy key material ASAP once it's no longer needed. I've added a TODO to DekCache for this. This might be difficult though, because it implies that the DekCache "knows" when all the DekContexts that it's returned have been used.

Reply:
Shouldn't the `DekContext` know when its being used to encrypt something and thus track its own usage? It could then refuse to encrypt /throw past expiry and get users to regenerate?